src/classes/frDonor.cls

```diff
@@ -102,22 +102,27 @@ public class frDonor extends frModel {
             String postalCode = String.valueOf(request.get('postalCode'));
             String country = String.valueOf(request.get('country'));
             
-            //try to match by address, in case we sync an offline donation over with no email.
-            String byAddress = 'select Id, fr_ID__c, LastName from Contact where MailingStreet = :address1' +
-                ' and MailingCity = :city' +
-                ' and MailingState = :state' +
-                ' and MailingPostalCode = :postalCode' +
-                ' and MailingCountry = :country' +
-                ' and FirstName = :firstName' +
-                ' and LastName = :lastName' +
-                ' limit 1 FOR UPDATE';
-            
-            contacts = (List<Contact>)Database.query(byAddress);
-            supporterContact = applyMatch(contacts, frId);
+            //name AND street address AND (cityAndState or postalCode) 
+            //need to be populated for us to match
+            Boolean namePresent = String.isNotBlank(firstName) && String.isNotBlank(lastName);
+            Boolean cityAndState = String.isNotBlank(city) && String.isNotBlank(state);
+            if(namePresent && String.isNotBlank(address1) && (cityAndState || String.isNotBlank(postalCode))) {
+                String byAddress = 'select Id, fr_ID__c, LastName from Contact where MailingStreet = :address1' +
+                    ' and MailingCity = :city' +
```

Comment:
I'm not sure about this thought, but is there anything to only querying on the fields that are in the request? If they include `name AND street address AND cityAndState` but not `postalCode`, should we exclude `postalCode` from the query?

Reply:
I really like that, I'm going to see if we can incorporate that logic without making the query code too messy